src/firefly/js/visualize/XYPlot.jsx

```diff
@@ -34,288 +35,610 @@ const plotParamsShape = PropTypes.shape({
+    yMax: PropTypes.number,
+    weightMin: PropTypes.number,
+    weightMax: PropTypes.number,
+    idStr: PropTypes.string
+});
+
+const UNSELECTED = 'unselected';
+const SELECTED = 'selected';
+const HIGHLIGHTED = 'highlighted';
+
 const unselectedColor = 'rgba(63, 127, 191, 0.5)';
 const selectedColor = 'rgba(21, 138, 15, 0.5)';
 const highlightedColor = 'rgba(250, 243, 40, 1)';
-const selectionRectColor = 'rgba(165, 165, 165, 0.25)';
-
-var XYPlot = React.createClass(
-    {
-        displayName: 'XYPlot',
-
-        propTypes: {
-            data: PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.string)), // array of numbers [0] - nInBin, [1] - binMin, [2] - binMax
-            width: PropTypes.number,
-            height: PropTypes.number,
-            params: plotParamsShape,
-            highlightedRow: PropTypes.number,
-            selectInfo: PropTypes.shape({
-                selectAll: PropTypes.bool,
-                exceptions: PropTypes.instanceOf(Set),
-                rowCount: PropTypes.number
-            }),
-            onHighlightChange: PropTypes.func,
-            onSelection: PropTypes.func,
-            desc: PropTypes.string
-        },
-
-        getDefaultProps() {
-            return {
-                data: undefined,
-                params: undefined,
-                highlightedRow: 0,
-                onHighlightChange: undefined,
-                onSelection: undefined,
-                height: 300,
-                desc: 'Sample XY Plot'
-            };
-        },
+const selectionRectColor = 'rgba(165, 165, 165, 0.5)';
 
-        componentDidMount() {
-            this.adjustPlotDisplay();
-        },
+const toNumber = (val)=>Number(val);
 
-        componentDidUpdate(pProps, pState, pContext) {
-            this.adjustPlotDisplay();
-        },
-
-        adjustPlotDisplay() {
-            const chart = this.refs.chart.getChart();
-            const {params} = this.props;
-
-            if (params.zoom) {
-                const {xMin, xMax, yMin, yMax} = params.zoom;
-                // redraw=true, animation=false
-                chart.xAxis[0].setExtremes(xMin, xMax, true, false);
-                chart.yAxis[0].setExtremes(yMin, yMax, true, false);
+/*
+ @param weight for a given point
+ @param minWeight minimum weight for all points
+ @param maxWeight maximum weigh for all points
+ @param logShading - if true use log color scale
+ @return{number} from 1 to 6, 1 for 1 pt series
+ */
+const getWeightBasedGroup = function(weight, minWeight, maxWeight, logShading=true) {
+    if (weight == 1) return 1; // 1pt;
+    else {
+        if (logShading) {
+            //use log scale for shade assignment
+            let max; //min=2, max;
+            const base = Math.pow(maxWeight+0.5, 0.2);
+            for (var e = 1; e <=5; e++) {
+                max = Math.round(Math.pow(base, e));
+                if (weight <= max) {
+                    if (max > maxWeight) { max = maxWeight; }
```

Comment:
I may be wrong. Just wondering what  this assignment (max = maxweight) would impact?